src/java/org/apache/cassandra/net/OutboundTcpConnection.java

```diff
@@ -507,7 +508,7 @@ private boolean connect()
                 // SSL errors won't be recoverable within timeout period so we'll just abort
                 return false;
             }
-            catch (IOException e)
+            catch (Exception e)
```

Comment:
Replacing IOException w/ Exception is stemming from `SSLFactory.maybeConnectWithTimeout()`. The call to `Future.get()` throws InterruptedException, ExecutionException, and TimeoutException. Then the call to `Callable.call` throws `Exception`. This is what we discussed offline in wrapping w/ a "mega exception" rather than propagating each of these up the stack.